lang/test/org/partiql/lang/eval/EvaluatingCompilerTests.kt

```diff
@@ -1475,4 +1535,68 @@ class EvaluatingCompilerTests : EvaluatorTestBase() {
                      listOf("a", "b"),
                      result.first().orderedNames)
     }
+
+    @Test
+    fun selectDistinct() = assertEval(
+        """SELECT DISTINCT t.a FROM `[{a: 1}, {a: 2}, {a: 1}]` t""",
+        """
+          [{a: 1}, {a: 2}]
+        """)
+
+    @Test
+    fun selectDistinctWithAggregate() = assertEval(
+        """SELECT SUM(DISTINCT t.a) AS a FROM `[{a:10}, {a:1}, {a:10}, {a:3}]` t""",
+        "[{a:14}]")
+
+    @Test
+    fun selectDistinctSubQuery() = assertEval(
+        """SELECT * FROM (SELECT DISTINCT t.a FROM `[{a: 1}, {a: 2}, {a: 1}]` t)""", 
+        """[{a:1},{a:2}]""")
+    
+    @Test
+    fun selectDistinctWithGroupBy() = assertEval(
+        """
+            SELECT t.a, COUNT(DISTINCT t.b) AS c 
+            FROM `[{a:1, b:10}, {a:1, b:10}, {a:1, b:20}, {a:2, b:10}, {a:2, b:10}]` t
+            GROUP by t.a
+        """,
+        """[{a:1, c:2}, {a:2, c:1}]""")
+
+    @Test
+    fun selectDistinctWithJoin() = assertEval(
+        """
+            SELECT DISTINCT *  
+            FROM 
+                `[1, 1, 1, 1, 2]` t1,
+                `[2, 2, 2, 2, 1]` t2
+        """,
+        """
+          [{_1:1,_2:2}, {_1:1, _2:1}, {_1:2,_2:2}, {_1:2,_2:1}]
+        """)
+
+    @Test
+    fun selectDistinctStar() = assertEval(
+        """
+            SELECT DISTINCT * 
+            FROM [
+                1, 1, 2, 
+                [1], [1], [1, 2], 
+                <<>>, <<>>, 
+                MISSING, NULL, NULL, MISSING, 
+                {'a':1}, {'a':1}, {'a':2}]
+        """,
+        """
+          [{_1:1},{_1:2},{_1:[1]},{_1:[1,2]},{_1:[]},{},{_1:null},{a:1},{a:2}]
```

Comment:
Thanks for the pointer `assertExprEquals` looks like the better, vs `assertEval` option in all cases really. 

Unfortunately `assertExprEquals` doesn't have any javadoc and is not as used as `assertEval` so it's easy to miss it. I'm thinking of marking `assertEval` as deprecated and adding documentation so we slowly move towards the better assertion. Let me know if you have any concerns

Reply:
>  I'm thinking of marking assertEval as deprecated and adding documentation so we slowly move towards the better assertion.
+1

> We might want to consider expressing expected results using the PTS syntax instead of ExprValue.ionValue this way we can clearly see in the error message if something was (missing) instead of null and if a sequence is a (bag ...) instead of a list [...]

You might want to make an override of `assertExprEquals` that accepts an SQL string for the expected value--then you can parse and evaluate the SQL in order to obtain the `ExprValue`.  `EvaluatingCompilerGroupByTest` does something like that already.

Reply:
Actually looking a bit more into it I don't think we should use `assertExprEquals` in this level of testing `EvaluatingCompilerTests` has end to end tests (lexer, parser, compiler) and to define the expected result we'd need to either: 
* Use the tree components under testing to parse a string with the expected `ExprValue` which seems bad as we are using what we are testing to define the expectation 
* Manually construct `ExprValue`s using the factory which would make the test horrible to read...